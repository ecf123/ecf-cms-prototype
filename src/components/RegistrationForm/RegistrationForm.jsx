import LoginInputBox from "../LoginInputBox/LoginInputBox";
import "./RegistrationForm.scss";

const RegistrationForm = () => {
  return (
    <div className="registration-form">
      <h2 className="registration-form__header">Register</h2>
      <form className="registration-form__form">
        <LoginInputBox
          labelText="First Name"
          inputType="text"
          labelId="first-name"
        />
        <LoginInputBox
          labelText="Last Name"
          inputType="text"
          labelId="last-name"
        />
        <LoginInputBox labelText="Email" inputType="email" labelId="email" />
        <LoginInputBox
          labelText="Password"
          inputType="password"
          labelId="password"
        />
        <LoginInputBox
          labelText="Re-Type Password"
          inputType="password"
          labelId="re-type-password"
        />
        <button className="registration-form__button">Create Account</button>
      </form>
    </div>
  );
};

export default RegistrationForm;
