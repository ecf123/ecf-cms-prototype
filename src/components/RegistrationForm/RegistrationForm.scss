@use "../../styles/variables" as *;

.registration-form {
  width: 22.125rem;

  &__header {
    color: $color-black;
    font-family: $font-secondary;
    font-size: $font-size-extra-large;
    margin-bottom: 2rem;
    text-align: center;
  }

  &__label {
    display: block;
    font-family: $font-secondary;
    font-size: $font-size-extra-small;
  }

  &__button {
    width: 100%;
    background-color: $color-green;
    height: 2.875rem;
    border: none;
    border-radius: 0.4rem;
    font-size: $font-size-small;
    font-family: $font-secondary;
    margin-top: 1rem;
    cursor: pointer;
  }
}
