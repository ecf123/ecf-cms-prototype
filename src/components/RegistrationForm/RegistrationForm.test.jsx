import { customRender } from "../../utils/testUtils";
import RegistrationForm from "./RegistrationForm";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

it("it should match the snapshot", () => {
  const { container } = customRender(<RegistrationForm />);
  expect(container).toMatchSnapshot();
});

it("Should render labels", () => {
  render(<RegistrationForm />);

  const labelOne = screen.getByText("First Name");
  const labelTwo = screen.getByText("Last Name");

  expect(labelOne).toBeInTheDocument();
  expect(labelTwo).toBeInTheDocument();
});

it("should not display text when password entered", () => {
  render(<RegistrationForm />);

  const passwordInput = screen.getByLabelText("Password");
  userEvent.type(passwordInput, "fix");

  const password = screen.queryByText("fix");
  expect(password).toBeFalsy();
});
