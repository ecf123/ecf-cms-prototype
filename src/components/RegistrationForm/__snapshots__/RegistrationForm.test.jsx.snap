// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot 1`] = `
<div>
  <div
    class="registration-form"
  >
    <h2
      class="registration-form__header"
    >
      Register
    </h2>
    <form
      class="registration-form__form"
    >
      <div
        class="input-box"
      >
        <label
          class="input-box__label"
          for="first-name"
        >
          First Name
        </label>
        <input
          alt="input box"
          class="input-box__input"
          id="first-name"
          required=""
          type="text"
        />
      </div>
      <div
        class="input-box"
      >
        <label
          class="input-box__label"
          for="last-name"
        >
          Last Name
        </label>
        <input
          alt="input box"
          class="input-box__input"
          id="last-name"
          required=""
          type="text"
        />
      </div>
      <div
        class="input-box"
      >
        <label
          class="input-box__label"
          for="email"
        >
          Email
        </label>
        <input
          alt="input box"
          class="input-box__input"
          id="email"
          required=""
          type="email"
        />
      </div>
      <div
        class="input-box"
      >
        <label
          class="input-box__label"
          for="password"
        >
          Password
        </label>
        <input
          alt="input box"
          class="input-box__input"
          id="password"
          required=""
          type="password"
        />
      </div>
      <div
        class="input-box"
      >
        <label
          class="input-box__label"
          for="re-type-password"
        >
          Re-Type Password
        </label>
        <input
          alt="input box"
          class="input-box__input"
          id="re-type-password"
          required=""
          type="password"
        />
      </div>
      <button
        class="registration-form__button"
      >
        Create Account
      </button>
    </form>
  </div>
</div>
`;
